PennMobile/Courses/CourseScheduleViewController.swift

```diff
@@ -56,18 +34,143 @@ extension CourseScheduleViewController {
         courseScheduleTableView = ModularTableView()
         courseScheduleTableView.backgroundColor = .clear
         courseScheduleTableView.separatorStyle = .none
-        
+
         view.addSubview(courseScheduleTableView)
-        
+
         courseScheduleTableView.anchorToTop(nil, left: view.leftAnchor, bottom: nil, right: view.rightAnchor)
         courseScheduleTableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 0).isActive = true
         courseScheduleTableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: 0).isActive = true
         courseScheduleTableView.tableFooterView = UIView(frame: CGRect(x: 0.0, y: 0.0, width: self.view.frame.width, height: 30.0))
-        
+
         HomeItemTypes.instance.registerCells(for: courseScheduleTableView)
     }
 }
 
+// MARK: - Refresh Controller
+extension CourseScheduleViewController {
+    func prepareRefreshControl() {
+        courseScheduleTableView.refreshControl = UIRefreshControl()
+        courseScheduleTableView.refreshControl?.addTarget(self, action: #selector(handleCourseRefresh), for: .valueChanged)
+    }
+
+    @objc func handleCourseRefresh() {
+        PennInTouchNetworkManager.instance.getCourses(currentTermOnly: true, callback: { (result) in
+            DispatchQueue.main.async {
+                self.handleNetworkCourseRefreshResult(result)
+                self.courseScheduleTableView.refreshControl?.endRefreshing()
+            }
+        })
+    }
+
+    // TODO: Need review: ad-hoc method to overcome conflict of UIAlertController and UIRefreshController
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        self.courseScheduleTableView.isHidden = true
+        self.courseScheduleTableView.refreshControl?.endRefreshing()
+        self.courseScheduleTableView.contentOffset = CGPoint.zero
```

Comment:
This is an ad-hoc method.
For some reason, UIAlertController and UIRefreshController seems to be in conflict with one another that prevents me from seeing the effect of calling `endRefreshing()`

Similar issues has been brought up by other [people](https://stackoverflow.com/questions/40230892/uirefresh-control-endrefreshing-doesnt-work). I wasn't sure if you knew of a better work around way of doing this.